uisdk/src/main/java/tech/dojo/pay/uisdk/domain/mapper/PaymentIntentDomainEntityMapper.kt

```diff
@@ -3,57 +3,78 @@ package tech.dojo.pay.uisdk.domain.mapper
-                PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.AUTHORIZED,
-        )
+    fun apply(raw: PaymentIntentPayload): PaymentIntentDomainEntity? {
+        return if (checkInvalidParameters(raw)) {
+            null
+        } else {
+            PaymentIntentDomainEntity(
+                id = requireNotNull(raw.id),
+                customerId = raw.customer?.id,
+                paymentToken = requireNotNull(raw.clientSessionSecret),
+                amount = AmountDomainEntity(
+                    valueLong = requireNotNull(raw.amount?.value ?: raw.intendedAmount?.value),
+                    valueString = requireNotNull(
+                        raw.amount?.value?.centsToString()
+                            ?: raw.intendedAmount?.value?.centsToString(),
+                    ),
+                    currencyCode = requireNotNull(
+                        raw.amount?.currencyCode ?: raw.intendedAmount?.currencyCode,
+                    ),
+                ),
+                supportedCardsSchemes = requireNotNull(raw.merchantConfig?.supportedPaymentMethods?.cardSchemes?.mapNotNull { it }),
+                supportedWalletSchemes = raw.merchantConfig?.supportedPaymentMethods?.wallets
+                    ?: emptyList(),
+                itemLines = raw.itemLines?.map {
+                    ItemLinesDomainEntity(
+                        amount = it.amountTotal,
+                        caption = it.caption,
+                    )
+                },
+                collectionEmailRequired = raw.config?.customerEmail?.collectionRequired ?: false,
+                collectionBillingAddressRequired = raw.config?.billingAddress?.collectionRequired
+                    ?: false,
+                collectionShippingAddressRequired = raw.config?.shippingDetails?.collectionRequired
+                    ?: false,
+                isVirtualTerminalPayment = raw.paymentSource?.let { it.lowercase() == "virtual-terminal" }
+                    ?: false,
+                isPreAuthPayment = raw.captureMode?.let { it.lowercase() == "manual" } ?: false,
+                orderId = raw.reference ?: "",
+                isSetUpIntentPayment = !raw.merchantInitiatedType.isNullOrBlank() && !raw.paymentSource.isNullOrBlank(),
+                merchantName = raw.config?.tradingName ?: "",
+                isPaymentAlreadyCollected =
+                PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.CAPTURED ||
+                    PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.AUTHORIZED,
+            )
+        }
     }
 
-    private fun checkInvalidParameters(raw: PaymentIntentPayload) {
+    private fun checkInvalidParameters(raw: PaymentIntentPayload): Boolean {
```

Comment:
Just being picky, I think it can be renamed to `containsInvalidParameters`, because `checkInvalidParameters` doesn't refer to a boolean so it turns a bit hard to understand the result 😅